### PR TITLE
Minor improvements to health check messages and formatting

### PR DESCRIPTION
**Pull request by danielfoehrKn**
**How to categorize this PR?**

/area quality
/kind technical-debt

**What this PR does / why we need it**:

Includes multiple minor improvements to health check detail messages by the Gardenlet and also in the Extension health check library.
- consistently use quotation marks instead of 'xyz' to indicate values

- make failing condition status more human readable. Exclude the type (Available) and the expected status (True).
  - ` condition "Available" has invalid status False (expected True) due to MinimumReplicasUnavailable: Deployment does not have minimum availability`
    becomes `Deployment does not have minimum availability (MinimumReplicasUnavailable)`
- replace health check summary with reader-friendlier alternative 
  - `Health check summary: 1/3 unsuccessful, 0/3 progressing, 2/3 successful `
    becomes `1 failing health check`.
- `"Worker" CRD reports failed ...`. --> `Worker extension reports failed ...`

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator

```

### CODE REVIEW

File: extensions/pkg/controller/healthcheck/general/managed_resource.go
Comment (schrodit) [inline extensions/pkg/controller/healthcheck/general/managed_resource.go]: nit: maybe I would make sense to use the new error syntax introduced with go1.13 (https://blog.golang.org/go1.13-errors). `%w` acts exactly like `%v` but the differnece that the initial error could be unwrapped if needed.
```suggestion
		err := fmt.Errorf("managed resource %q in namespace %q is unhealthy: %w", managedResource.Name, managedResource.Namespace, err)
```

File: pkg/utils/kubernetes/health/health.go
Comment (ialidzhikov) [inline pkg/utils/kubernetes/health/health.go]: Hmm, why this is handled in this way? The Shoot conditions are already respecting thresholds which are configurable with the `controllers.shootCare.conditionThresholds` field in the gardenlet component config. So `EveryNodeReady` condition will be `Processing` for the configured threshold. Hence I don't see a reason why we introduce such hard-coded threshold.

Comment (danielfoehrKn) [reply pkg/utils/kubernetes/health/health.go]: valid point. I saw these temporary unhealthy nodes during scaling up & drain operations.
I have to investigate this further. Thanks for raising the concern.

Comment (danielfoehrKn) [reply pkg/utils/kubernetes/health/health.go]: Could not reproduce it. I removed it. Will come back to it when I see it again.

File: extensions/pkg/controller/healthcheck/general/daemonset.go
Comment (rfranzke) [inline extensions/pkg/controller/healthcheck/general/daemonset.go:123]: Doc string

**Review by rfranzke — COMMENTED** [review]
I haven't found much, though, I wasn't able to assess everything. I guess multiple commits would have helped...

**Review by rfranzke — DISMISSED** [review]
/lgtm but I would like to get a 
/second-opinion

**Comment by rfranzke** [comment]
ping @gardener/gardener-maintainers

File: extensions/pkg/controller/healthcheck/general/daemonset.go
Comment (ialidzhikov) [inline extensions/pkg/controller/healthcheck/general/daemonset.go]: Nit - you could move to the 3rd import group.

```
	"context"
	"fmt"

	"github.com/gardener/gardener/extensions/pkg/controller/healthcheck"
	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"

	"github.com/gardener/gardener/pkg/utils/kubernetes/health"
	"github.com/go-logr/logr"
	appsv1 "k8s.io/api/apps/v1"
	apierrors "k8s.io/apimachinery/pkg/api/errors"
	"k8s.io/apimachinery/pkg/types"
	"sigs.k8s.io/controller-runtime/pkg/client"
	"sigs.k8s.io/controller-runtime/pkg/log"
```

File: extensions/pkg/controller/healthcheck/general/deployment.go
Comment (ialidzhikov) [inline extensions/pkg/controller/healthcheck/general/deployment.go]: Nit - you could move to the 3rd import group.

```
	"context"
	"fmt"

	"github.com/gardener/gardener/extensions/pkg/controller/healthcheck"
	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"

	"github.com/gardener/gardener/pkg/utils/kubernetes/health"
	"github.com/go-logr/logr"
	appsv1 "k8s.io/api/apps/v1"
	apierrors "k8s.io/apimachinery/pkg/api/errors"
	"k8s.io/apimachinery/pkg/types"
	"sigs.k8s.io/controller-runtime/pkg/client"
	"sigs.k8s.io/controller-runtime/pkg/log"
```

File: extensions/pkg/controller/healthcheck/general/managed_resource.go
Comment (ialidzhikov) [inline extensions/pkg/controller/healthcheck/general/managed_resource.go]: Nit - you could move to the 3rd import group.

File: extensions/pkg/controller/healthcheck/general/statefulsets.go
Comment (ialidzhikov) [inline extensions/pkg/controller/healthcheck/general/statefulsets.go]: Nit - you could move to the 3rd import group.

**Review by ialidzhikov — COMMENTED** [review]
Minor nits, otherwise lgtm.

**Review by timuthy — COMMENTED** [review]
Looks good to me code wise, but out of curiosity, did you take a provider extension and tried to revendor your changes? I wondered if we can have a quick trial run before merging and releasing this.

**Comment by danielfoehrKn** [comment]
> Looks good to me code wise, but out of curiosity, did you take a provider extension and tried to revendor your changes? I wondered if we can have a quick trial run before merging and releasing this.

Yes I did. However, it was now already a while ago. I'll update the PR, revendor and deploy it again.

**Comment by danielfoehrKn** [comment]
@timuthy I updated the PR. I send you a PN .

**Review by timuthy — APPROVED** [review]
/lgtm

**Pull request merged**
